app/modules/sightings/resources.py

```diff
@@ -57,21 +158,180 @@ def get(self, args):
+            log.warning(
+                '_validate_asset_references threw %r on encounters=%r'
+                % (ex, request_in['encounters'])
+            )
+            _cleanup_post_and_abort(
+                result_data['id'], None, 'Invalid assetReference data in encounter(s)'
+            )
+        log.debug(
+            '_validate_asset_references returned: %r, %r' % (all_arefs, paths_wanted)
+        )
+
+        submission = None
+        assets_added = None
+        pub = True
+        owner_guid = None
+        if current_user is not None and not current_user.is_anonymous:
+            owner_guid = current_user.guid
+            pub = False
+
+        if all_arefs is not None and paths_wanted is not None:
+            # files seem to exist in uploads dir, so lets make
+            transaction_id = next(
+                iter(all_arefs)
+            )  # here is where we make single-transaciton-id assumption
+            from app.modules.submissions.models import Submission
+
+            try:
+                submission = Submission.create_submission_from_tus(
+                    'Sighting.post ' + result_data['id'],
+                    current_user,
+                    transaction_id,
+                    paths=all_arefs[transaction_id],
+                )
+            except Exception as ex:
+                log.error(
+                    '%r on create_submission_from_tus transaction_id=%r paths=%r'
+                    % (
+                        ex,
+                        transaction_id,
+                        all_arefs[transaction_id],
+                    )
+                )
+                _cleanup_post_and_abort(
+                    result_data['id'], submission, 'Problem with encounter/assets'
+                )
+
+            assets_added = submission.assets
+            log.debug(
+                'create_submission_from_tus returned: %r => %r'
+                % (submission, assets_added)
+            )
+
+        sighting = Sighting(
+            guid=result_data['id'],
+            version=result_data.get('version', 2),
+        )
+
+        # create encounters (including adding their assets if applicable)
+        from app.modules.encounters.models import Encounter
+
+        if isinstance(result_data['encounters'], list):
+            i = 0
+            while i < len(result_data['encounters']):
+                try:
+                    encounter = Encounter(
+                        guid=result_data['encounters'][i]['id'],
+                        version=result_data['encounters'][i].get('version', 2),
+                        owner_guid=owner_guid,
+                        public=pub,
+                    )
+                    enc_assets = None
+                    if paths_wanted is not None:
+                        # will throw exception if we dont have all we need
+                        enc_assets = _enc_assets(assets_added, paths_wanted[i])
+                    if enc_assets is not None:
```

Comment:
Just wondering if it would be clearer if this line and the next two were within the 
if paths_wanted is not None:

Reply:
you mean cuz `enc_assets` can only be set within that block?  yeah i guess it technically saves a cpu cycle or 2 also.  will change.